### ISSUE TITLE
Android App - Set Up Encryption fails

### ISSUE THREAD

**Issue by Wizlonex** [nextcloud/android#2013]
### Actual behaviour
Fails to setup encryption option on android app.

### Expected behaviour
N/A
 
### Steps to reproduce
1. Select folder to encrypt.  
2. Tap menu.  
3. Select "Set as encrypted".  
4. Click folder.  
    --> "Note your encryption passphrase...".  
5. Click "Set Up Encryption".  
   --> Error message "Error.  Storing keys was unsuccessful, please try it again!".
6. Click close.
   --> (If you try to enter the "encrypted" folder, the process (and error) repeats from step 4).

If I try to access the encrypted folder via the web portal, it errors (probably expected behavior?).  

Nothing useful seen in server log because it does not record the client-side error (but it does record the accessing encrypted folder failure, which is normal I think?).  I am happy to try to provide additional info.

If I try to send app logs, the app crashes ('Nextcloud has stopped').  But even so, there are no app log entries associated with the encryption set-up failure anyway.

### Environment data
Android version: 8.1.0

Device model: Pixel XL 2

Stock or customized system: Stock

Nextcloud app version:  3.0.0 RC1

Nextcloud server version: 13.0.0 RC1

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
@schiessle strange that there is nothing on server log, or?
I assumed that the CSR is not valid, maybe related to #1859

**Comment by ConorIA** [comment]
I'm experiencing the same behaviour.

```
Error
end_to_end_encryption
2018-01-23T17:15:18-0500
Can't create public key: could not sign the CSR, please make sure to submit a valid CSR

Error
PHP
2018-01-23T17:15:18-0500
openssl_csr_sign(): cannot get CSR from parameter 1 at /apps2/end_to_end_encryption/lib/SignatureHandler.php#60
```

**Comment by ConorIA** [comment]
Not sure if relevant, but I'm using this Docker image: https://github.com/hoellen/dockerfiles/blob/master/nextcloud/Dockerfile.13.0

Could this be a libressl issue? (I'm really not sure about cross compatibility between open and libre.)

**Comment by ardevd** [comment]
Having the same issue on my Pixel 2 XL with the latest version of the Nextcloud app from Google Play. See #2134

**Comment by ConorIA** [comment]
Not fixed as of 3.0.1 for me. Still getting the same errors in the logs: 

```
Error }     end_to_end_encryption | Can't create public key: could not sign the CSR, please make sure to submit a valid CSR | 2018-02-23T10:33:19-0500
-- | -- | -- | --
Error | PHP | openssl_csr_sign(): cannot get CSR from parameter 1 at /apps2/end_to_end_encryption/lib/SignatureHandler.php#60 | 2018-02-23T10:33:19-0500

Error | end_to_end_encryption | Can't create public key: could not sign the CSR, please make sure to submit a valid CSR | 2018-02-23T10:33:19-0500
-- | -- | -- | --
Error | PHP | openssl_csr_sign(): cannot get CSR from parameter 1 at /apps2/end_to_end_encryption/lib/SignatureHandler.php#60 | 2018-02-23T10:33:19-0500
```
